File: lab 10.cpp
#include <iostream>
using namespace std;

class GPA_Calculator {
public:
    float calculateGPA(int marks[], int numSubjects) {
        float total = 0.0;
        for (int i = 0; i < numSubjects; i++) {
            total += marks[i];
        }
        return total / numSubjects / 10.0; // Assuming the GPA scale is out of 10
    }
};

int main() {
    int numSubjects;

    cout << "Enter the number of subjects: ";
    cin >> numSubjects;

    int marks[numSubjects];
    for (int i = 0; i < numSubjects; i++) {
        cout << "Enter marks for subject " << i + 1 << ": ";
        cin >> marks[i];
    }

    GPA_Calculator gpaCalculator;
    float gpa = gpaCalculator.calculateGPA(marks, numSubjects);

    cout << "Your GPA is: " << gpa << endl;

    return 0;
}

File: lab 10.java
10. Write a program that takes marks in individual subjects from the user and calculate the GPA.
( By C++ or Java )



import java.util.Scanner;

class GPACalculator {

    public float calculateGPA(int[] marks, int numSubjects) {
        float total = 0;
        for (int mark : marks) {
            total += mark;
        }
        return total / numSubjects / 10.0f; // Assuming the GPA scale is out of 10
    }
}

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter the number of subjects: ");
        int numSubjects = sc.nextInt();

        int[] marks = new int[numSubjects];
        for (int i = 0; i < numSubjects; i++) {
            System.out.print("Enter marks for subject " + (i + 1) + ": ");
            marks[i] = sc.nextInt();
        }

        GPACalculator gpaCalculator = new GPACalculator();
        float gpa = gpaCalculator.calculateGPA(marks, numSubjects);

        System.out.println("Your GPA is: " + gpa);

        sc.close();
    }
}
